vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/search/AbstractElementSearchAction.java

```diff
@@ -64,26 +71,26 @@
             if (elementsFound)
             {
                 Visibility visibility = parameters.getVisibility();
-                return Visibility.ALL == visibility ? elements : filterElementsByVisibility(elements,
+                try
+                {
+                    return Visibility.ALL == visibility ? elements : filterElementsByVisibility(elements,
                         visibility == Visibility.VISIBLE);
+                } catch (StaleElementReferenceException e) {
+                    if (retrySearchIfStale && !isRetry) {
+                        return findElements(searchContext, locator, parameters, true);
+                    }
+                    LOGGER.warn(e.getMessage(), e);
```

Comment:
logic is broken here: if `retrySearchIfStale` is `false`, then first check of element visibility throwing `StaleElementReferenceException` will result in empty list, but we just need to have filtered out list